### PR TITLE
Update documentation for provider December 2021 release

### PR DESCRIPTION
**Pull request by potiuk**
---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/main/UPDATING.md).

### CODE REVIEW

**Comment by eladkal** [comment]
So it's too late for https://github.com/apache/airflow/pull/20514 :)

**Review by eladkal — APPROVED** [review]

File: airflow/providers/amazon/CHANGELOG.rst [hunk 1]
```diff
@@ -19,6 +19,56 @@
...
+* ``Add sensor for AWS Batch (#19850) (#19885)``
+* ``Add state details to EMR container failure reason (#19579)``
+
+Bug Fixes
+~~~~~~~~~
+
+* ``Fix backwards compatibility issue in AWS provider's _get_credentials (#20463)``
+* ``Fix deprecation messages after splitting redshift modules (#20366)``
+* ``ECSOperator: fix KeyError on missing exitCode (#20264)``
+* ``Bug fix in AWS glue operator when specifying the WorkerType &amp; NumberOfWorkers (#19787)``
```
Comment (eladkal) [inline airflow/providers/amazon/CHANGELOG.rst]: Should it have `&amp;` ?

**Comment by potiuk** [comment]
> So it's too late for #20514 :)

Not too late . We can add it still. I think it makes sense actually as we have breaking changes there already so we can add this one if we merge it soon.

File: airflow/providers/amazon/CHANGELOG.rst (hunk 1 above)
Comment (potiuk) [reply airflow/providers/amazon/CHANGELOG.rst]: Not really :). But I can use the #20514 to fix it.

File: airflow/providers/snowflake/CHANGELOG.rst [hunk 2]
```diff
@@ -19,6 +19,20 @@
...
+
+Features
+~~~~~~~~
+
+* ``Support insecure mode in SnowflakeHook (#20106)``
+* ``Remove unused code in SnowflakeHook (#20107)``
+
+.. Below changes are excluded from the changelog. Move them to
+   appropriate section above if needed. Do not delete the lines(!):
+   * ``Fix MyPy Errors for Snowflake provider. (#20212)``
```
Comment (mik-laj) [inline airflow/providers/snowflake/CHANGELOG.rst:36]: 3 changes that were made to the Snowflake provider are missing but I don't know if this is a problem.
https://github.com/apache/airflow/pull/20245
https://github.com/apache/airflow/pull/20395
https://github.com/apache/airflow/pull/20095

Comment (potiuk) [reply airflow/providers/snowflake/CHANGELOG.rst:36]: Good you noitced. The first 2  are "setup.py" changes and they are not - as such - visible in provider changes. There is currently no link between such "setup.py-only" changes and providers - but we can add them manually. The last one is test-only change and those are not included (and should not be).

File: airflow/providers/amazon/CHANGELOG.rst (hunk 1 above)
Comment (potiuk) [reply airflow/providers/amazon/CHANGELOG.rst]: Fixed it permanently (missing "| safe" in Jinja filter.

File: airflow/providers/snowflake/CHANGELOG.rst (hunk 2 above)
Comment (potiuk) [reply airflow/providers/snowflake/CHANGELOG.rst:36]: Updated manually.

**Review by mik-laj — APPROVED** [review]

**Pull request merged**